testpar/testphdf5.c

```diff
@@ -346,7 +346,19 @@ main(int argc, char **argv)
     test_params.count = 0;
 
     /* Tests are generally arranged from least to most complexity... */
+
+    /* Skip this test for OpenMPI version 5.0.5 since an issue was
+     * introduced in that release which causes this test to fail.
+     * The issue was fixed for the 5.0.6 release.
+     */
+#if defined(OPEN_MPI) && defined(OMPI_MAJOR_VERSION) && (OMPI_MAJOR_VERSION == 5) &&                         \
+    defined(OMPI_MINOR_VERSION) && (OMPI_MINOR_VERSION == 0) && defined(OMPI_RELEASE_VERSION) &&             \
```

Comment:
It would probably make sense to have a macro for MPI library version checking in case we need to do the same for an MPICH release in the future, but it didn't seem too worth it yet.